### PR TITLE
executor: make unit test TestInvalidReadTemporaryTable more stable

### LINKED ISSUE #32950: UT unstable test TestInvalidReadTemporaryTable
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_unit_test/detail/tidb_ghpr_unit_test/39655/pipeline/

```
[2022-03-09T07:06:30.740Z] --- FAIL: TestInvalidReadTemporaryTable (0.49s)
[2022-03-09T07:06:30.740Z]     result.go:50: 
[2022-03-09T07:06:30.740Z]         	Error Trace:	result.go:50
[2022-03-09T07:06:30.740Z]         	            				executor_txn_test.go:122
[2022-03-09T07:06:30.740Z]         	Error:      	Not equal: 
[2022-03-09T07:06:30.740Z]         	            	expected: "[1]\n"
[2022-03-09T07:06:30.740Z]         	            	actual  : ""
[2022-03-09T07:06:30.741Z]         	            	
[2022-03-09T07:06:30.741Z]         	            	Diff:
[2022-03-09T07:06:30.741Z]         	            	--- Expected
[2022-03-09T07:06:30.741Z]         	            	+++ Actual
[2022-03-09T07:06:30.741Z]         	            	@@ -1,2 +1 @@
[2022-03-09T07:06:30.741Z]         	            	-[1]
[2022-03-09T07:06:30.741Z]         	            	 
[2022-03-09T07:06:30.741Z]         	Test:       	TestInvalidReadTemporaryTable
[2022-03-09T07:06:30.741Z]         	Messages:   	sql:select * from tmp6 as of timestamp(@a) where id=1;, args:[]
[2022-03-09T07:06:30.741Z] FAIL
```

### 2. What did you expect to see? (Required)

CI pass

### 3. What did you see instead (Required)

Unstable test

### 4. What is your TiDB version? (Required)

master 4ab66b8ac2c55716384e1360fcd566ea6fa3a35e

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #32950

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [X] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

File: executor/executor_txn_test.go [hunk 1]
```diff
@@ -110,13 +110,13 @@ func TestInvalidReadTemporaryTable(t *testing.T) {
 	tk.MustExec("commit")
 
 	for _, query := range queries {
-		tk.MustExec(query.sql)
+		tk.MustQuery(query.sql)
```
Comment (tisonkun) [inline executor/executor_txn_test.go:113]: This change seems changing semantic. I'm curious how this change hardens the test?

Comment (tiancaiamao) [reply executor/executor_txn_test.go:113]: The result is not checked... I'm not modifing the logic here.

Comment (tisonkun) [reply executor/executor_txn_test.go:113]: Fine. It can be `MustExec` though.

**Pull request merged**